bootstrap/kubeadm/controllers/kubeadmconfig_controller.go

```diff
@@ -596,3 +607,42 @@ func (r *KubeadmConfigReconciler) reconcileTopLevelObjectSettings(cluster *clust
 		log.Info("Altering ClusterConfiguration", "KubernetesVersion", config.Spec.ClusterConfiguration.KubernetesVersion)
 	}
 }
+
+// storeBootstrapData creates a new secret with the data passed in as input,
+// sets the reference in the configuration status and ready to true.
+func (r *KubeadmConfigReconciler) storeBootstrapData(ctx context.Context, config *bootstrapv1.KubeadmConfig, data []byte) error {
+	secret := &corev1.Secret{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      config.Name,
+			Namespace: config.Namespace,
+			Labels: map[string]string{
+				clusterv1.ClusterLabelName: config.ClusterName,
+			},
+			OwnerReferences: []v1.OwnerReference{
+				{
+					APIVersion: bootstrapv1.GroupVersion.String(),
+					Kind:       "KubeadmConfig",
+					Name:       config.Name,
+					UID:        config.UID,
+					Controller: pointer.BoolPtr(true),
+				},
+			},
+		},
+		Data: map[string][]byte{
+			"value": data,
+		},
+	}
+
+	key := types.NamespacedName{Name: secret.Name, Namespace: secret.Namespace}
+	if err := r.Client.Get(ctx, key, &corev1.Secret{}); apierrors.IsNotFound(err) {
+		if err := r.Client.Create(ctx, secret); err != nil {
+			return err
```

Comment:
wrap?

Reply:
Callers wraps

Reply:
We need the wrap as low in the stack as possible. If you wait for the caller to wrap, you lose that it came from this line.

Reply:
You're right. Fixed.